src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/ReadsDataflowSource.java

```diff
@@ -142,6 +143,22 @@ public SAMFileHeader getHeader() {
         return preads;
     }
 
+    /**
+      * Create a {@link PCollection< KV <String,Iterable<GATKRead>>>} containing all the reads starting in the given intervals and their mates,
+      * grouped by the shard they start in.
+      * The shards are strings as per SimpleInterval's ToString syntax.
```

Comment:
Don't use the `toString()` representation as the key (`toString()` is just for display purposes) -- use the `SimpleIntervals` themselves.